### ISSUE TITLE
apiURL isn't fully used, apart from detecting for localhost, clientId, & branch

### ISSUE THREAD

**Issue by jamespohalloran** [tinacms/tinacms#2439]
As a user (developer) I want to be able to configure apiURL so that I can change my local api port, or use a custom Content API URL so that I can manage multiple sites running on my local machine.

**Expected Results:**
When setting `apiURL` on the Tina config:
- If the apiURL doesn't match the `https://content.tinajs.io/content/<client-id>/github/<branch>` regex, we use the apiURL as the content API URL directly. (E.g, localhost:4001, https://my-custom-url.blah, etc)
- Show a nice error message if it's not localhost or content.tinajs.io

**Notes:**

Our TinaCMS container has an apiURL prop, and it's currently used for detecting isLocalClient, and grabbing the clientId & branch.  https://github.com/tinacms/tinacms/blob/main/packages/tinacms/src/tina-cms.tsx#L294
It's a bit deceiving however, as the actual URL isn't used apart from grabbing those properties.

#### Tinacms Package Versions

```
tinacms: 0.63.0
```

**Comment by jamespohalloran** [comment]
Duplicate of #2433

**Comment by jamespohalloran** [comment]
Whoops! This is a separate issue from its marked duplicate. Reopening

**Comment by logan-anderson** [comment]
> If the apiURL doesn't match the above regex, we use the apiURL as the content API URL directly. (E.g, localhost:4001, https://my-custom-url.blah/, etc)

Right now we have two "modes" local and production.  Local has no auth and just uses the content API (one endpoint). Production requires an auth endpoint, a dashboard endpoint (LOGIN UI) and a content API. For this work to be completed do we need a third mode? or are we OK with when a user provides a customURL they are in local mode and see Local mode banner, ect?

**Comment by jamespohalloran** [comment]
It looks like "local-mode" (which controls that local banner), is controlled by the presence of "localhost" in the url. I think that logic can stay the same. https://github.com/tinacms/tinacms/blob/main/packages/tinacms/src/client/index.ts#L122

While there are technically 3 possible states right now, I'm still thinking of it as a single apiURL, with a few sub-states driven off of its value:
- localmode (true|false) is determined by apiURL containing "localhost"
- tinaCloud (true|false) (I don't think this is an actual named variable, but it controls the branch switcher stuff) is controlled by apiURL matching "https://content.tinajs.io/content/<client-id>/github/<branch>"

With the above thinking, I would assume that `apiURL: 'https://fake-url.blah'` would not show the banner.

Does that make sense @logan-anderson ?

**Comment by dwalkr** [comment]
> While there are technically 3 possible states right now, I'm still thinking of it as a single apiURL, with a few sub-states driven off of its value:

I currently feel that our intention to derive these operating states from a single `apiUrl` value is causing unnecessary headaches here. I don't recall why we adopted the `apiUrl` approach in the first place, but it's become clear that it's a very leaky abstraction. Ultimately I think the content API should be configured via a different paradigm than a single URL string, but I recognize that's a broader issue that I don't think can be adequately accomplished until we bring extensibility back to Tina.

I point out the above because I think it reinforces how tightly we should control the scope on this issue. Any solution that continues to derive configuration by parsing `apiUrl` is, in my mind, temporary. That is of course fine, but if that's the case then we can focus this issue on unblocking users who are blocked by the current behavior.

**Comment by jamespohalloran** [comment]
The gist of the original issue was that we had several props that were conditional on the values of other props being set. 
- isLocalClient
- clientId - (only makes sense when not in local-mode)
- branch - (only makes sense when not in local-mode)

I do agree things probably need a bigger revisit though. apiUrl was introduced before we have a branching plugin, and our picture of how self-hosting will work is definetely changing. 

I pulled this task out of the current sprint, as the "3 points" estimate was a bit too high for the level of value that it brings right now.
